client/state/signup/steps/website-content/reducer.ts

```diff
@@ -37,10 +45,24 @@ export default withSchemaValidation(
 					currentIndex: action.payload,
 				};
 
-			case SIGNUP_STEPS_WEBSITE_CONTENT_IMAGE_UPLOADED: {
+			case SIGNUP_STEPS_WEBSITE_CONTENT_IMAGE_UPLOAD_STARTED: {
 				const { payload } = action;
 
-				const pageIndex = state.websiteContent.findIndex( ( page ) => page.id === payload.id );
+				return {
```

Comment:
I was wondering if it's required to store the upload state for every `pageId`/`mediaIndex`? We need to block navigation if any image upload is in progress, so a simple boolean flag can also work here. Similar to this reducer:
https://github.com/Automattic/wp-calypso/blob/fff000dd11e0a0fb37d9b285ae872c5ea9ca7fc5/client/state/sites/reducer.js#L282-L293

Reply:
Actually the main motivation for this is to allow multiple images to be upload within a section. Also if we want to allow switching between sections, or show an upload progress indicated in a section while images are being uploaded as a UX enhancement having the section vise state will be beneficial.